include/foxy/detail/export_connect_fields.hpp

```diff
@@ -97,25 +92,21 @@ foxy::detail::export_connect_fields(Fields& src, Fields& dst)
   // iterate the `src` fields, moving any connect headers and the
   // corresponding tokens to the `dst` fields
   //
-  auto const hop_by_hops =
-    std::array<http::field, 11>{http::field::connection,
-                                http::field::keep_alive,
-                                http::field::proxy_authenticate,
-                                http::field::proxy_authentication_info,
-                                http::field::proxy_authorization,
-                                http::field::proxy_connection,
-                                http::field::proxy_features,
-                                http::field::proxy_instruction,
-                                http::field::te,
-                                http::field::trailer,
-                                http::field::transfer_encoding};
-
-  auto const is_connect_opt =
-    [&connect_opts,
-     &hop_by_hops](typename Fields::value_type const& field) -> bool {
-    if (range::find(hop_by_hops, field.name()) != hop_by_hops.end()) {
-      return true;
-    }
+  auto const hop_by_hops = std::array<http::field, 11>{http::field::connection,
+                                                       http::field::keep_alive,
+                                                       http::field::proxy_authenticate,
+                                                       http::field::proxy_authentication_info,
+                                                       http::field::proxy_authorization,
+                                                       http::field::proxy_connection,
+                                                       http::field::proxy_features,
+                                                       http::field::proxy_instruction,
+                                                       http::field::te,
+                                                       http::field::trailer,
+                                                       http::field::transfer_encoding};
+
+  auto const is_connect_opt = [&connect_opts,
+                               &hop_by_hops](typename Fields::value_type const& field) -> bool {
+    if (range::find(hop_by_hops, field.name()) != hop_by_hops.end()) { return true; }
```

Comment:
There is no guarantee that the `http::field` values will be in any particular order, so a binary search won't work.

Reply:
I don't understand this. As far as I'm aware, `range::find` does a linear traversal through the range.